src/Synchronizer.js

```diff
@@ -88,11 +88,38 @@ module.exports = Class.extend({
          })
          .catch(function(err) {
             self._abortScanning = true;
-            console.log('ERROR: encountered an error while comparing tables', err, err.stack);
+            console.error('Encountered an error while comparing tables', err, err.stack);
          })
          .then(this._outputStats.bind(this));
    },
 
+   trackScanProgress: function(enabled, approxItems, f) {
+      var tick = approxItems > 100 ? 1 : 5,
+          currentPercent = 0,
+          nextProgressPercent = 0,
+          ret;
+
+
+      ret = f(function(currentItems) {
+         currentPercent = Math.round(currentItems / approxItems * 100.0);
+
+         if (currentPercent < 100 && currentPercent >= (nextProgressPercent + tick)) {
+            process.stdout.write('(' + currentPercent + '%)');
+
+            nextProgressPercent =
+               currentPercent >= (nextProgressPercent + tick) ?
+               (nextProgressPercent + tick) :
+               (currentPercent + tick);
+         } else {
+            process.stdout.write('.');
+         }
+      });
+
+      return ret.then(function() {
+         process.stdout.write('(100%)');
```

Comment:
and just flagging one more spot that would need something

Reply:
this should probably include a '\n' however that works to be consistent.